### PR TITLE
Add support for ERA5-land

### LINKED ISSUE #54: Add support for ERA5 back extention and ERA5-Land
ERA5 now has a preliminary back extension (1950-1978).

As it has a seprate entry in the CDS, we need to teach era5cli how to download this and other era5 variants such as ERA5-Land

See here for an overview: https://confluence.ecmwf.int/display/CKB/The+family+of+ERA5+datasets

### PR DESCRIPTION
**Pull request by Peter9192**
This pull request adds functionality for downloading the ERA5-land datasets ([hourly](https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-land) and [monthly](https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-land-monthly-means)). For example: 

```bash
era5cli info land
```
will list all the available variables in the land data, and

```bash
era5cli monthly --variables snow_cover --startyear 2008 --land --dryrun
```
will print the request for 2008 monthly snow cover data (remove the dry run to actually download the data). 

Documentation has been added, as well as several new checks and tests. As the number of input parameters grows, we believe it would be useful to comprehensively check the compatibility of all the available options. We opened a new issue to address this (#68).

closes #54

### CODE REVIEW

File: era5cli/fetch.py [hunk 1]
```diff
@@ -59,17 +59,20 @@ class Fetch:
...
                  hours: list, variables: list, outputformat: str,
                  outputprefix: str, period: str, ensemble: bool,
                  statistics=None, synoptic=None, pressurelevels=None,
-                 merge=False, threads=None, prelimbe=False):
+                 merge=False, threads=None, prelimbe=False, land=False):
         """Initialization of Fetch class."""
         self.months = era5cli.utils._zpad_months(months)
         """list(str): List of zero-padded strings of months
         (e.g. ['01', '02',..., '12'])."""
+        # TODO make sure these docstrings are consistent with cli docs
```
Comment (nielsdrost) [inline era5cli/fetch.py:75]: Does this still need doing? :-)

File: tests/test_fetch.py [hunk 2]
```diff
@@ -201,79 +209,208 @@ def test_number_outputfiles(capsys):
...
+
+    # But with the right dataset option it should pass
+    era5.land = True
+    era5._check_variable(land_only_variable)
+
+    # And variables incompatible with ERA5 land should fail
+    with pytest.raises(ValueError):
+        era5._check_variable(slev_only_variable)
+
+    # Missing monthly vars should pass if period is hourly
```
Comment (nielsdrost) [inline tests/test_fetch.py:357]: perhaps set the period explicitly here? I think this currently depends on the default period setting, right?

**Review by nielsdrost — CHANGES_REQUESTED** [review]
Nice work @bvreede and @Peter9192 !

A few small comments, otherwise good to go :-)

File: era5cli/fetch.py (hunk 1 above)
Comment (bvreede) [reply era5cli/fetch.py:75]: Yes, but it is probably better done in a separate pull request.

Comment (bvreede) [reply era5cli/fetch.py:75]: Addressing the docstring consistency may also fit addressing issue #68, and ensuring input is verified comprehensively. It is beyond the scope of this PR.

Comment (bvreede) [reply era5cli/fetch.py:75]: Added an issue #71 and removing the #TODO :)

**Review by nielsdrost — APPROVED** [review]

**Comment by nielsdrost** [comment]
Thanks @bvreede!

**Pull request merged**